src/realm/array.cpp

```diff
@@ -1717,9 +1717,9 @@ void Array::alloc(size_t init_size, size_t width)
             // field in the header.
             size_t new_capacity_bytes = capacity_bytes * 2;
             if (new_capacity_bytes < capacity_bytes) // overflow detected, clamp to max
```

Comment:
Just for the record, I think this check is perfectly fine. It shows the intend perfectly clear.